### DISCUSSION TITLE
JavaVers Connector to Elastic Search

### DISCUSSION THREAD

**Discussion by PHMT-CW** [javers/javers#1144] [General]
Hi, I'm very interested in this library and it's capability for acting as a Repository for commits, etc... can you help me understand how much effort it would be to make an ElasticSearch connector? 
- Could the MongoDB connector be adapted to work with ElasticSearch? 
- Has anyone started working on this connector already? 
- How long did it take to create the MongoDB connector?

**Comment by bartoszwalacik** [comment]
Hi, take a look at https://github.com/javers/javers/blob/master/javers-persistence-mongo/src/main/java/org/javers/repository/mongo/MongoRepository.java
You can take this code as an example and write sth similar for Elastic. Generally speaking all you have to do is to implement the [JaversRepository](https://github.com/javers/javers/blob/master/javers-core/src/main/java/org/javers/repository/api/JaversRepository.java) interface.

If you write it, think about contributing this Elastic integration to Javers.

**Reply by PHMT-CW** [reply]
Thanks for the quick reply, I'll take a look. If we do implement something successfully for Elastic, I'll consider sharing it here.
